### PR TITLE
Manual Ratings

### PR DESCRIPTION
**Pull request by tknorris**
This gets manual ratings working for movies, shows, and episodes. Note, even though trakt 2.0 supports season ratings, I didn't include them here because the original script call didn't. Let me know if you want me to add them in. 

Note, this PR is dependent on the script.module.trakt PR I'm about to submit as well.

I only tested manual ratings and no other functions. Once you confirm that I should, I'll do another pull to push a beta for this and the script.module.trakt changes.

### CODE REVIEW

**Comment by tknorris** [comment]
Note, I couldn't come up w/ an easy way to remove all those beta commits.

**Comment by s7eele** [comment]
awesome, thx

**Comment by Razzeee** [comment]
Hey awesome work. 
Will have a look soon and merge. I would also want to include the season ratings. 
Maybe we should also add something to the readme about these calls?

**Comment by tknorris** [comment]
On that DBID code, that might be useless code but I didn't really check. In the old addon that code was there because the tagging functionality needed it after a rating was submitted. If that tagging code has been removed, then it's also probably that the DBID isn't used any more as well. I didn't take the chance of removing it though because I wasn't sure and the cost of assigning an extra key to the dict was minimal.

**Comment by tknorris** [comment]
I'll start taking a look at what is needed to make seasons ratings work. I think it's going to take quite a few changes.

**Comment by Razzeee** [comment]
Just had a look at the sync. From a quick check, seems like it's still working as before.

**Comment by tknorris** [comment]
Great, let me know when you are ready and after you do a merge I can do a pull of both repositories and push them to my beta repo.

**Comment by s7eele** [comment]
im kinda late, want me to test anything

**Comment by Razzeee** [comment]
Well I'm waiting for the season ratings?
Also we should update the changelog and maybe provide a section in the readme documenting the available calls?

**Comment by tknorris** [comment]
Sorry, I thought we were going to do the season ratings next. I'll take a look at them this weekend.

On the readme there is already this: https://github.com/trakt/script.trakt/wiki/Action%3A-rate in the script.trakt wiki. Do we need something in the wiki as well? I'll update the change log when I submit the  PR for seasons ratings.

**Comment by Razzeee** [comment]
We'll I was aware of that, but maybe we should mention the wiki in the documentation?

**Comment by Razzeee** [comment]
I'll merge this now. Seasons are probably just nice to have, I'll probably get to work on markWatched later.
Trying to bring back the contextMenu is also on my agenda, but the code seems pretty "uncool". Not really sure.

**Comment by tknorris** [comment]
We'll do whatever you want. You've done most of the work to get the addon working with Trakt 2.0. As far as I'm concerned, it's your baby. I'm just throwing some code your way that I use in SALTS. ;-)

**Comment by tknorris** [comment]
Do you want me to go ahead and do a beta for this and script.module.trakt?

**Comment by Razzeee** [comment]
Yep, let's do that.
And I don't really like that "your baby" definition. Everyone who can code is welcome to contribute.

**Comment by tknorris** [comment]
hahah, I'm just saying you are the one who took the initiative to get everything working and those of us who just sat around a waited have to be respectful of that.

**Comment by Razzeee** [comment]
Respect is good and everything, but I like teamwork and it's pretty bad if one person is maintaining a project. No vacations for me I guess ;)

**Comment by tknorris** [comment]
You mean you didn't realize you were committing yourself to a live of servitude when you updated the addon? Well, someone should have told you! ;-)

**Comment by Razzeee** [comment]
Well, somebody had to do it, but yeah - I guess I should have seen it comming...

And whatever I want is really sounding like you would accept really really bad code. Thats not what I want but I may not be aware of it, so it shouldn't be whatever I want. Because I want to learn and be corrected if it is nescessary.

**Comment by tknorris** [comment]
I think you already know this, but just to double check I was going to bump the version numbers on both of the two addons so that they are beyond the ones currently in the kodi repo. So, anyone with my beta repo will be running the beta now. Right?

**Comment by Razzeee** [comment]
Correct, but you've increased the version of the trakt module already

**Comment by freeman212** [comment]
@tknorris not until you push these changes to your beta repo of course ;-)

File: service.py [hunk 1]
```diff
@@ -139,23 +139,32 @@ def doManualRating(self, data):
...
+			userInfo = globals.traktapi.getShowRatingForUser(data['imdbnumber'], 'imdb')
 		elif utilities.isMovie(media_type):
 			summaryInfo = globals.traktapi.getMovieSummary(data['imdbnumber'])
+			userInfo = globals.traktapi.getMovieRatingForUser(data['imdbnumber'])
 		
-		if not summaryInfo is None:
-			if utilities.isMovie(media_type) or utilities.isShow(media_type):
+		if summaryInfo is not None:
+			summaryInfo = summaryInfo.to_dict()
+			if 'dbid' in data and (utilities.isMovie(media_type) or utilities.isShow(media_type)):
```
Comment (tknorris) [inline service.py]: Back on your comment about this code not being needed. One way the manual ratings can be called is w/ a library DBID passing in as "dbid". In which case this code would get used.

**Comment by Razzeee** [comment]
Are you sure? Can't find a place where xbmc_id is accessed.

**Comment by tknorris** [comment]
I'm sure that in 2.3.1 it was there but the code might have been lost in
3.0. If so, we should probably add it back as its a documented part of the
script.trakt interface. I'll look at that at the same time time I do the
season stuff this weekend.

**Comment by s7eele** [comment]
'do it at the same time' now your just bragging tknorris lol

**Pull request merged**